### DISCUSSION TITLE
How were the constants for the whitespace_table lookup in json_character_block::classify choosen?

### DISCUSSION THREAD

**Discussion by camel-cdr** [simdjson/simdjson#2290] [Q&A]
I'm a bit confused why such non-obvious constants were used in the `whitespace_table` lookup table:

https://github.com/simdjson/simdjson/blob/9b314922a1f5ecbda366115992893e368cd2f0e0/src/icelake.cpp#L51

https://github.com/simdjson/simdjson/blob/9b314922a1f5ecbda366115992893e368cd2f0e0/src/icelake.cpp#L84-L86

If I understood it correctly, the only criterion for the constants is that they don't contain their index in the lower four bits.
Wouldn't using zeros for all non-whitespace characters be more intuitive?

**Comment by lemire** [comment]
This code has been around a long time, and I am not certain where these unused numbers come from. There might be some historical reason. We could probably trace it back using the git history.

**Reply by camel-cdr** [reply]
It seems to originate from this commit: https://github.com/simdjson/simdjson/commit/6c168f046dc7234f0bb37d53778dd9d531b12dc0#diff-ae56130591292550ceef73821a56532805a46ee2f4115507103d2c7a1c5aebddR622

**Reply by lemire** [reply]
Yep, I think that's right, and you can see that the logic was different at the time.
